### ISSUE TITLE
Deep merge fails for an attribute marked with both @JsonMerge and @JsonUnwrapped

### ISSUE THREAD

**Issue by fraj** [FasterXML/jackson-annotations#148]
I have an abstract generic wrapper class `UnwrappedExtended<T>` that wraps an object of type `T` and has an additional text attribute named `additional`. `UnwrappedExtendedPerson` is a subclass of `UnwrappedExtended` where the wrapped object is of type `Person`. `Person`'s attributes are `firstName` and `lastName`.

I want to mark `UnwrappedExtended`'s attribute `basic` with annotation `@JsonUnwrapped` so that the JSON representation for an `UnwrappedExtendedPerson` will be e.g.
> {"firstName": "John", "lastName": "Doe", "additional": "Jay"}

I also want to mark attribute `basic` with annotation `@JsonMerge` so that I can patch an `UnwrappedExtendedPerson` with 
> {"firstName": "Jane", "additional": "X"}

 and the resulting object would be
> {"firstName": "Jane", "lastName": "Doe", "additional": "X"}

However, when the same attribute is marked with `@JsonUnwrapped` and `@JsonMerge`, the merge fails and the result is
> {"firstName": "Jane", "lastName": null, "additional": "X"}

Cf. attached project for classes and unit tests. Also contains `Extended<T>
[fraj-jackson-issue.zip](https://github.com/FasterXML/jackson-annotations/files/2851664/fraj-jackson-issue.zip)
` with basic attribute only marked with `@JsonMerge`.

**Comment by cowtowncoder** [comment]
Yes, `@JsonUnwrapped` does not play nice with many other features unfortunately.
I hope this can be improved with 3.0, but for Jackson 2.x it is possible this can not be addressed.

But I will transfer this to `jackson-databind` as this repo only declares annotations: actual handling is in databind code.
